core/src/main/java/org/jdbi/v3/core/config/ConfigRegistry.java

```diff
@@ -81,12 +81,45 @@ public <C extends JdbiConfig<C>> C get(Class<C> configClass) {
     }
 
     private Function<ConfigRegistry, JdbiConfig<?>> configFactory(Class<? extends JdbiConfig<?>> configClass) {
-        return configFactories.computeIfAbsent(configClass, klass ->
-                registry -> {
-                    var config = JdbiClassUtils.findConstructorAndCreateInstance(klass, JDBI_CONFIG_TYPES, registry);
-                    config.setRegistry(registry);
-                    return config;
-                });
+        return configFactories.computeIfAbsent(configClass, klass -> {
+            final Exception notFound;
+            try {
+                MethodHandle mh = MethodHandles.lookup().findConstructor(klass,
+                        MethodType.methodType(void.class, ConfigRegistry.class))
+                    .asType(MethodType.methodType(JdbiConfig.class, ConfigRegistry.class));
+                return registry -> {
```

Comment:
Unreadable is very much a style opinion. I moved the cache computation function to a helper method, so there are no nested lambdas now.

The sneaky throws were intentionally unrolled - previously, we used `Unchecked`, but this introduces more stack frames. I thought it was preferable to handle it locally rather than nesting more lambdas. Do you like the `Unchecked.function` style better?

Do you have further concrete suggestions for how it could be more readable?